src/networking.c

```diff
@@ -4954,7 +4954,9 @@ void ioThreadReadQueryFromClient(void *data) {
     }
 
 done:
-    trimClientQueryBuffer(c);
+    if (!(c->read_flags & READ_FLAGS_PRIMARY)) {
```

Comment:
I think this is worth a comment explaining why we don't want to trim the client query buffer. The special copying between buffers is not obvious.

Reply:
Done